apps/asap-server/src/handlers/webhooks/fetch-by-code/fetch-by-code.ts

```diff
@@ -0,0 +1,51 @@
+import { SearchClient } from 'algoliasearch';
+import Joi from '@hapi/joi';
+import { framework as lambda } from '@asap-hub/services-common';
+import { UserMetadataResponse } from '@asap-hub/model';
+import { http } from '../../../utils/instrumented-framework';
+import { UserController } from '../../../controllers/users';
+import validateRequest from '../../../utils/validate-auth0-request';
+import { Handler } from '../../../utils/types';
+import { algoliaSearchApiKey } from '../../../config';
+import logger from '../../../utils/logger';
+
+export const fetchUserByCodeHandlerFactory = (
+  userController: UserController,
+  algoliaClient: SearchClient,
+): Handler =>
+  http(async (request: lambda.Request): Promise<
+    lambda.Response<UserMetadataResponse>
+  > => {
+    await validateRequest(request);
+
+    const paramsSchema = Joi.object({
+      code: Joi.string().required(),
+    }).required();
+
+    const { code } = lambda.validate(
+      'params',
+      request.params,
+      paramsSchema,
+    ) as {
+      code: string;
+    };
+
+    const user = await userController.fetchByCode(code);
+    let apiKey: string | null;
+
+    try {
+      apiKey = algoliaClient.generateSecuredApiKey(algoliaSearchApiKey, {
+        validUntil: Date.now() + 36001, // which is one second over the TTL of the ID token
+      });
+    } catch (e) {
+      apiKey = null;
```

Comment:
I thought we agreed we would fail the auth request if getting the token fails. Having a user unable to browse listing pages is pretty critical. I was in favour of handling this on another endpoint in the frontend over it being possibly null

Reply:
we did and like i said on slack - because i cant test auth0 flow without it being deployed to dev (and so - prod) i want to use this first, merge, test and then remove the null - i specifically used null not undefined for this